src/coreComponents/physicsSolvers/fluidFlow/wells/CompositionalMultiphaseWell.cpp

```diff
@@ -863,12 +909,23 @@ void CompositionalMultiphaseWell::updatePhaseVolumeFraction( WellElementSubRegio
 
   string const & fluidName = subRegion.getReference< string >( viewKeyStruct::fluidNamesString() );
   MultiFluidBase & fluid = subRegion.getConstitutiveModel< MultiFluidBase >( fluidName );
-  isothermalCompositionalMultiphaseBaseKernels::
-    PhaseVolumeFractionKernelFactory::
-    createAndLaunch< parallelDevicePolicy<> >( m_numComponents,
-                                               m_numPhases,
-                                               subRegion,
-                                               fluid );
+
+  real64 maxDeltaPhaseVolFrac  =
+    m_isThermal ?
+    thermalCompositionalMultiphaseBaseKernels::
+      PhaseVolumeFractionKernelFactory::
+      createAndLaunch< parallelDevicePolicy<> >( m_numComponents,
+                                                 m_numPhases,
+                                                 subRegion,
+                                                 fluid )
+:    isothermalCompositionalMultiphaseBaseKernels::
+      PhaseVolumeFractionKernelFactory::
+      createAndLaunch< parallelDevicePolicy<> >( m_numComponents,
+                                                 m_numPhases,
+                                                 subRegion,
+                                                 fluid );
+
+  maxDeltaPhaseVolFrac = MpiWrapper::max( maxDeltaPhaseVolFrac );
```

Comment:
this should be better outside of `subRegion` section
we observed bad parallel degradation for flow solver when similar sync was placed inside subregion loop

Reply:
moved comm to outside subregion loop in latest push